### ISSUE TITLE
Not all Buckets shown from InfluxDB

### ISSUE THREAD

**Issue by svogel123** [grafana/grafana#47925]
When I setup an connection to my InfluxDB ist shows only 18 of 57 Buckets

Could anywhere help to show all the Buckets in grafana.

**Environment**:
- Grafana version: ~~v8.4.6 (c53173ff6)~~ --> now 8.5.4
- OS Grafana is installed on: Ubuntu 20.04.4
- User OS & Browser: Windows 10
- InfluxDB v2.2.0

**Comment by yesoreyeram** [comment]
Can you post the response you see in the query inspector of Grafana? or the response from the browser network tools.

**Comment by svogel123** [comment]
This are my buckets in InfluxDB
![2022-06-07_15h00_37](https://user-images.githubusercontent.com/104064222/172387707-a1ea7a11-b5ab-4b43-8339-750ac12b27d8.png)

When I Filter the Buckets in Parameters there are two Buckets missing with *mzd*
![2022-06-07_14h57_41](https://user-images.githubusercontent.com/104064222/172387710-bcac2d2b-3317-4f63-b078-79d7e9103ff7.png)

Here is the response of the inspector:
[response.txt](https://github.com/grafana/grafana/files/8853449/response.txt)

It is necessary to show all buckets in grafana.

**Comment by svogel123** [comment]
when i call the api (/api/v2/buckets) in the browser i get the following response:
[buckets_20_to_end.txt](https://github.com/grafana/grafana/files/8858515/buckets_20_to_end.txt)
[buckets_00_to_20.txt](https://github.com/grafana/grafana/files/8858516/buckets_00_to_20.txt)

with `curl --request GET "http://.../api/v2/buckets" --header "Authorization: Token ..."` i get the same response
[curl_buckets_00_to_20.txt](https://github.com/grafana/grafana/files/8858541/curl_buckets_00_to_20.txt)

**Comment by yesoreyeram** [comment]
looks like `/api/v2/buckets` is the different API. Grafana uses `/api/v1/query` api to get the results. 

Example : `curl -H 'Authorization: Token XXXXX' -H 'Content-Type: application/json' -X POST 'http://localhost:8086/api/v2/query?org=XXXX' -d '{"dialect":{"annotations":["datatype","group","default"],"delimiter":",","header":true},"query":"buckets()","type":"flux"}'` is the query grafana uses under the hood.

In theory, there shouldn't be any difference between those APIs. But I would like you to confirm.

**Comment by svogel123** [comment]
Hi Sriranm,
thank you for your fast Response.

When i query the data with the other API call. I get all Buckets.

Find details:
[curl_query_response.txt](https://github.com/grafana/grafana/files/8860174/curl_query_response.txt)

![2022-06-08_11h11_50](https://user-images.githubusercontent.com/104064222/172579826-6e9ac49a-1488-4912-9d3a-f336c5899b90.png)

**Comment by svogel123** [comment]
We have upgraded to 8.5.4

**Comment by yesoreyeram** [comment]
Thank you for the details. Still I am not able to reproduce the issue. With the data you shared, I am getting 9 buckets with the `/.*ai_dispatcher_mzd*./` filter applied. Also attached the exported dashboard JSON for reference.

<img width="795" alt="image" src="https://user-images.githubusercontent.com/153843/172593623-015964d9-6f9f-4670-b64a-8c90eb8deb1a.png">

<details>
<summary>Dashboard JSON</summary>
{
  "__inputs": [
    {
      "name": "DS_INFLUXDB-BUCKETS-DEBUG",
      "label": "InfluxDB-Buckets-Debug",
      "description": "",
      "type": "datasource",
      "pluginId": "influxdb",
      "pluginName": "InfluxDB"
    }
  ],
  "__elements": [],
  "__requires": [
    {
      "type": "grafana",
      "id": "grafana",
      "name": "Grafana",
      "version": "9.0.0-3c3039f5pre"
    },
    {
      "type": "datasource",
      "id": "influxdb",
      "name": "InfluxDB",
      "version": "1.0.0"
    },
    {
      "type": "panel",
      "id": "table",
      "name": "Table",
      "version": ""
    }
  ],
  "annotations": {
    "list": []
  },
  "editable": true,
  "fiscalYearStartMonth": 0,
  "graphTooltip": 0,
  "id": null,
  "iteration": 1654683680093,
  "links": [],
  "liveNow": false,
  "panels": [
    {
      "datasource": {
        "type": "influxdb",
        "uid": "${DS_INFLUXDB-BUCKETS-DEBUG}"
      },
      "fieldConfig": {
        "defaults": {
          "color": {
            "mode": "thresholds"
          },
          "custom": {
            "align": "auto",
            "displayMode": "auto",
            "inspect": false
          },
          "mappings": [],
          "thresholds": {
            "mode": "absolute",
            "steps": [
              {
                "color": "green",
                "value": null
              },
              {
                "color": "red",
                "value": 80
              }
            ]
          }
        },
        "overrides": []
      },
      "gridPos": {
        "h": 22,
        "w": 24,
        "x": 0,
        "y": 0
      },
      "id": 2,
      "options": {
        "footer": {
          "fields": "",
          "reducer": [
            "sum"
          ],
          "show": false
        },
        "showHeader": true
      },
      "pluginVersion": "9.0.0-3c3039f5pre",
      "targets": [
        {
          "datasource": {
            "type": "influxdb",
            "uid": "${DS_INFLUXDB-BUCKETS-DEBUG}"
          },
          "query": "buckets()",
          "refId": "A"
        }
      ],
      "title": "Panel Title",
      "type": "table"
    }
  ],
  "schemaVersion": 36,
  "style": "dark",
  "tags": [],
  "templating": {
    "list": [
      {
        "allValue": "ai*",
        "current": {},
        "datasource": {
          "type": "influxdb",
          "uid": "${DS_INFLUXDB-BUCKETS-DEBUG}"
        },
        "definition": "buckets()",
        "hide": 0,
        "includeAll": true,
        "multi": true,
        "name": "bucket",
        "options": [],
        "query": "buckets()",
        "refresh": 1,
        "regex": "/.*ai_dispatcher_mzd*./",
        "skipUrlSync": false,
        "sort": 2,
        "type": "query"
      }
    ]
  },
  "time": {
    "from": "now-6h",
    "to": "now"
  },
  "timepicker": {},
  "timezone": "",
  "title": "Influx Bucket Debug",
  "uid": "2Pyx7GCnz",
  "version": 1,
  "weekStart": ""
}
</details>

I would suggest to import the above dashboard against your datasource and see if that makes any difference. Also make sure you use the same credentials in curl and grafana.

**Comment by svogel123** [comment]
I can't Import the Dashboard, do you use Grafana 9.0?


<details><summary>My Dashboard is:</summary>
`
{
  "annotations": {
    "list": [
      {
        "builtIn": 1,
        "datasource": {
          "type": "datasource",
          "uid": "grafana"
        },
        "enable": true,
        "hide": true,
        "iconColor": "rgba(0, 211, 255, 1)",
        "name": "Annotations & Alerts",
        "target": {
          "limit": 100,
          "matchAny": false,
          "tags": [],
          "type": "dashboard"
        },
        "type": "dashboard"
      }
    ]
  },
  "editable": true,
  "fiscalYearStartMonth": 0,
  "graphTooltip": 0,
  "id": 14,
  "iteration": 1654687858900,
  "links": [],
  "liveNow": false,
  "panels": [
    {
      "collapsed": false,
      "datasource": {
        "type": "influxdb",
        "uid": "yfZzmc1nk"
      },
      "gridPos": {
        "h": 1,
        "w": 24,
        "x": 0,
        "y": 0
      },
      "id": 229,
      "panels": [],
      "repeat": "Testrun",
      "title": "Details Ping - ${Testrun}",
      "type": "row"
    },
    {
      "datasource": {
        "type": "influxdb",
        "uid": "yfZzmc1nk"
      },
      "description": "",
      "gridPos": {
        "h": 2,
        "w": 24,
        "x": 0,
        "y": 1
      },
      "id": 340,
      "options": {
        "content": "<h1 style=\"text-align: center; background-color: blue\">${Testrun}</h1>\r\n",
        "mode": "html"
      },
      "pluginVersion": "8.5.4",
      "type": "text"
    },
    {
      "datasource": {
        "type": "influxdb",
        "uid": "yfZzmc1nk"
      },
      "fieldConfig": {
        "defaults": {
          "color": {
            "mode": "thresholds"
          },
          "decimals": 0,
          "mappings": [],
          "max": 60,
          "min": 0,
          "thresholds": {
            "mode": "absolute",
            "steps": [
              {
                "color": "green",
                "value": null
              }
            ]
          },
          "unit": "none"
        },
        "overrides": []
      },
      "gridPos": {
        "h": 6,
        "w": 24,
        "x": 0,
        "y": 3
      },
      "id": 213,
      "options": {
        "displayMode": "gradient",
        "minVizHeight": 10,
        "minVizWidth": 0,
        "orientation": "vertical",
        "reduceOptions": {
          "calcs": [
            "lastNotNull"
          ],
          "fields": "",
          "values": false
        },
        "showUnfilled": false
      },
      "pluginVersion": "8.5.4",
      "targets": [
        {
          "datasource": {
            "type": "influxdb",
            "uid": "yfZzmc1nk"
          },
          "query": "from(bucket: \"${Testrun}\")\r\n  |> range(start: v.timeRangeStart, stop: v.timeRangeStop)\r\n  |> filter(fn: (r) => r[\"_measurement\"] == \"ping\")\r\n  |> filter(fn: (r) => r[\"_field\"] == \"packetloss\")\r\n  |> group(columns: [\"source\"])\r\n  |> aggregateWindow(every: 1000h, fn: count, createEmpty: false)\r\n  |> yield(name: \"count\")",
          "refId": "A"
        }
      ],
      "title": "Ping count per Source",
      "transformations": [],
      "type": "bargauge"
    },
    {
      "datasource": {
        "type": "influxdb",
        "uid": "yfZzmc1nk"
      },
      "fieldConfig": {
        "defaults": {
          "color": {
            "mode": "thresholds"
          },
          "mappings": [],
          "max": 100,
          "thresholds": {
            "mode": "absolute",
            "steps": [
              {
                "color": "green",
                "value": null
              },
              {
                "color": "red",
                "value": 1
              }
            ]
          }
        },
        "overrides": []
      },
      "gridPos": {
        "h": 5,
        "w": 24,
        "x": 0,
        "y": 9
      },
      "id": 232,
      "links": [
        {
          "targetBlank": true,
          "title": "Show details",
          "url": "http://grafana.wtk.home:3000/d/LBcxZzfnk/mzd-ping-test-1-neu?orgId=1&var-Testrun=${Testrun}&var-source=All&var-destination=All"
        }
      ],
      "options": {
        "displayMode": "lcd",
        "minVizHeight": 10,
        "minVizWidth": 0,
        "orientation": "horizontal",
        "reduceOptions": {
          "calcs": [
            "lastNotNull"
          ],
          "fields": "",
          "values": false
        },
        "showUnfilled": true
      },
      "pluginVersion": "8.5.4",
      "targets": [
        {
          "datasource": {
            "type": "influxdb",
            "uid": "yfZzmc1nk"
          },
          "query": "from(bucket: \"${Testrun}\")\r\n  |> range(start: v.timeRangeStart, stop: v.timeRangeStop)\r\n  |> filter(fn: (r) => r[\"_measurement\"] == \"ping\")\r\n  |> filter(fn: (r) => r[\"_field\"] == \"packetloss\")\r\n  |> group(columns: [\"source\"])\r\n  |> aggregateWindow(every: 100m, fn: sum, createEmpty: false)\r\n  |> yield(name: \"sum\")",
          "refId": "A"
        }
      ],
      "title": "Packetloss in Percent",
      "transformations": [],
      "type": "bargauge"
    },
    {
      "datasource": {
        "type": "influxdb",
        "uid": "yfZzmc1nk"
      },
      "fieldConfig": {
        "defaults": {
          "color": {
            "mode": "thresholds"
          },
          "decimals": 3,
          "mappings": [],
          "min": 0,
          "thresholds": {
            "mode": "absolute",
            "steps": [
              {
                "color": "green",
                "value": null
              },
              {
                "color": "#EAB839",
                "value": 200
              },
              {
                "color": "red",
                "value": 250
              }
            ]
          },
          "unit": "ms"
        },
        "overrides": []
      },
      "gridPos": {
        "h": 6,
        "w": 12,
        "x": 0,
        "y": 14
      },
      "id": 336,
      "options": {
        "orientation": "auto",
        "reduceOptions": {
          "calcs": [
            "max"
          ],
          "fields": "",
          "values": false
        },
        "showThresholdLabels": false,
        "showThresholdMarkers": true
      },
      "pluginVersion": "8.5.4",
      "targets": [
        {
          "datasource": {
            "type": "influxdb",
            "uid": "yfZzmc1nk"
          },
          "query": "from(bucket: \"${Testrun}\")\r\n  |> range(start: v.timeRangeStart, stop: v.timeRangeStop)\r\n  |> filter(fn: (r) => r[\"_measurement\"] == \"ping\")\r\n  |> filter(fn: (r) => r[\"_field\"] == \"max\")\r\n  |> group(columns: [\"source\"])\r\n  |> aggregateWindow(every: 100m, fn: mean, createEmpty: false)\r\n  |> yield(name: \"mean\")",
          "refId": "A"
        }
      ],
      "title": "Max (mean - ms)",
      "transformations": [],
      "type": "gauge"
    },
    {
      "datasource": {
        "type": "influxdb",
        "uid": "yfZzmc1nk"
      },
      "fieldConfig": {
        "defaults": {
          "color": {
            "mode": "thresholds"
          },
          "decimals": 3,
          "mappings": [],
          "min": 0,
          "thresholds": {
            "mode": "absolute",
            "steps": [
              {
                "color": "green",
                "value": null
              },
              {
                "color": "#EAB839",
                "value": 200
              },
              {
                "color": "red",
                "value": 250
              }
            ]
          },
          "unit": "ms"
        },
        "overrides": []
      },
      "gridPos": {
        "h": 6,
        "w": 12,
        "x": 12,
        "y": 14
      },
      "id": 337,
      "options": {
        "orientation": "auto",
        "reduceOptions": {
          "calcs": [
            "min"
          ],
          "fields": "",
          "values": false
        },
        "showThresholdLabels": false,
        "showThresholdMarkers": true
      },
      "pluginVersion": "8.5.4",
      "targets": [
        {
          "datasource": {
            "type": "influxdb",
            "uid": "yfZzmc1nk"
          },
          "query": "from(bucket: \"${Testrun}\")\r\n  |> range(start: v.timeRangeStart, stop: v.timeRangeStop)\r\n  |> filter(fn: (r) => r[\"_measurement\"] == \"ping\")\r\n  |> filter(fn: (r) => r[\"_field\"] == \"min\")\r\n  |> group(columns: [\"source\"])\r\n  |> aggregateWindow(every: 100m, fn: mean, createEmpty: false)\r\n  |> yield(name: \"mean\")",
          "refId": "A"
        }
      ],
      "title": "min (ms)",
      "transformations": [],
      "type": "gauge"
    },
    {
      "datasource": {
        "type": "influxdb",
        "uid": "yfZzmc1nk"
      },
      "fieldConfig": {
        "defaults": {
          "color": {
            "mode": "thresholds"
          },
          "decimals": 3,
          "mappings": [],
          "min": 0,
          "thresholds": {
            "mode": "absolute",
            "steps": [
              {
                "color": "green",
                "value": null
              },
              {
                "color": "#EAB839",
                "value": 100
              },
              {
                "color": "red",
                "value": 150
              }
            ]
          },
          "unit": "ms"
        },
        "overrides": []
      },
      "gridPos": {
        "h": 6,
        "w": 12,
        "x": 0,
        "y": 20
      },
      "id": 335,
      "options": {
        "orientation": "auto",
        "reduceOptions": {
          "calcs": [
            "mean"
          ],
          "fields": "",
          "values": false
        },
        "showThresholdLabels": false,
        "showThresholdMarkers": true
      },
      "pluginVersion": "8.5.4",
      "targets": [
        {
          "datasource": {
            "type": "influxdb",
            "uid": "yfZzmc1nk"
          },
          "query": "from(bucket: \"${Testrun}\")\r\n  |> range(start: v.timeRangeStart, stop: v.timeRangeStop)\r\n  |> filter(fn: (r) => r[\"_measurement\"] == \"ping\")\r\n  |> filter(fn: (r) => r[\"_field\"] == \"avg\")\r\n  |> group(columns: [\"source\"])\r\n  |> aggregateWindow(every: 100m, fn: mean, createEmpty: false)\r\n  |> yield(name: \"mean\")",
          "refId": "A"
        }
      ],
      "title": "avg (ms)",
      "transformations": [],
      "type": "gauge"
    },
    {
      "datasource": {
        "type": "influxdb",
        "uid": "yfZzmc1nk"
      },
      "fieldConfig": {
        "defaults": {
          "color": {
            "mode": "thresholds"
          },
          "decimals": 3,
          "mappings": [],
          "min": 0,
          "thresholds": {
            "mode": "absolute",
            "steps": [
              {
                "color": "green",
                "value": null
              },
              {
                "color": "#EAB839",
                "value": 200
              },
              {
                "color": "red",
                "value": 250
              }
            ]
          },
          "unit": "ms"
        },
        "overrides": []
      },
      "gridPos": {
        "h": 6,
        "w": 12,
        "x": 12,
        "y": 20
      },
      "id": 338,
      "options": {
        "orientation": "auto",
        "reduceOptions": {
          "calcs": [
            "mean"
          ],
          "fields": "",
          "values": false
        },
        "showThresholdLabels": false,
        "showThresholdMarkers": true
      },
      "pluginVersion": "8.5.4",
      "targets": [
        {
          "datasource": {
            "type": "influxdb",
            "uid": "yfZzmc1nk"
          },
          "query": "from(bucket: \"${Testrun}\")\r\n  |> range(start: v.timeRangeStart, stop: v.timeRangeStop)\r\n  |> filter(fn: (r) => r[\"_measurement\"] == \"ping\")\r\n  |> filter(fn: (r) => r[\"_field\"] == \"mdev\")\r\n  |> group(columns: [\"source\"])\r\n  |> aggregateWindow(every: 100m, fn: mean, createEmpty: false)\r\n  |> yield(name: \"mean\")",
          "refId": "A"
        }
      ],
      "title": "mdev (ms)",
      "transformations": [],
      "type": "gauge"
    },
    {
      "datasource": {
        "type": "influxdb",
        "uid": "yfZzmc1nk"
      },
      "fieldConfig": {
        "defaults": {
          "color": {
            "mode": "thresholds"
          },
          "decimals": 0,
          "mappings": [],
          "max": 60,
          "min": 0,
          "thresholds": {
            "mode": "absolute",
            "steps": [
              {
                "color": "green",
                "value": null
              }
            ]
          },
          "unit": "none"
        },
        "overrides": []
      },
      "gridPos": {
        "h": 6,
        "w": 24,
        "x": 0,
        "y": 26
      },
      "id": 373,
      "options": {
        "displayMode": "gradient",
        "minVizHeight": 10,
        "minVizWidth": 0,
        "orientation": "vertical",
        "reduceOptions": {
          "calcs": [
            "lastNotNull"
          ],
          "fields": "",
          "values": false
        },
        "showUnfilled": false
      },
      "pluginVersion": "8.5.4",
      "targets": [
        {
          "datasource": {
            "type": "influxdb",
            "uid": "yfZzmc1nk"
          },
          "query": "from(bucket: \"${Testrun}\")\r\n  |> range(start: v.timeRangeStart, stop: v.timeRangeStop)\r\n  |> filter(fn: (r) => r[\"_measurement\"] == \"ftp\")\r\n  |> filter(fn: (r) => r[\"_field\"] == \"size\")\r\n  |> group(columns: [\"source\"])\r\n  |> aggregateWindow(every: 100m, fn: count, createEmpty: false)\r\n  |> yield(name: \"count\")",
          "refId": "A"
        }
      ],
      "title": "FTP count per Source",
      "transformations": [],
      "type": "bargauge"
    },
    {
      "datasource": {
        "type": "influxdb",
        "uid": "yfZzmc1nk"
      },
      "fieldConfig": {
        "defaults": {
          "color": {
            "mode": "thresholds"
          },
          "decimals": 3,
          "mappings": [],
          "min": 0,
          "thresholds": {
            "mode": "absolute",
            "steps": [
              {
                "color": "green",
                "value": null
              },
              {
                "color": "#EAB839",
                "value": 10
              },
              {
                "color": "red",
                "value": 25
              }
            ]
          },
          "unit": "ms"
        },
        "overrides": []
      },
      "gridPos": {
        "h": 8,
        "w": 12,
        "x": 6,
        "y": 32
      },
      "id": 496,
      "options": {
        "orientation": "auto",
        "reduceOptions": {
          "calcs": [
            "mean"
          ],
          "fields": "",
          "values": false
        },
        "showThresholdLabels": false,
        "showThresholdMarkers": true
      },
      "pluginVersion": "8.5.4",
      "targets": [
        {
          "datasource": {
            "type": "influxdb",
            "uid": "yfZzmc1nk"
          },
          "query": "from(bucket: \"${Testrun}\")\r\n  |> range(start: v.timeRangeStart, stop: v.timeRangeStop)\r\n  |> filter(fn: (r) => r[\"_measurement\"] == \"ftp\")\r\n  |> filter(fn: (r) => r[\"_field\"] == \"duration\")\r\n  |> keep(columns: [\"source\", \"duration\", \"_time\", \"_value\"])\r\n  |> group(columns: [\"source\"])",
          "refId": "A"
        }
      ],
      "title": "mean (duration)",
      "transformations": [],
      "type": "gauge"
    },
    {
      "datasource": {
        "type": "influxdb",
        "uid": "yfZzmc1nk"
      },
      "fieldConfig": {
        "defaults": {
          "color": {
            "mode": "thresholds"
          },
          "decimals": 3,
          "mappings": [],
          "min": 0,
          "thresholds": {
            "mode": "absolute",
            "steps": [
              {
                "color": "green"
              },
              {
                "color": "#EAB839",
                "value": 10
              },
              {
                "color": "red",
                "value": 25
              }
            ]
          },
          "unit": "ms"
        },
        "overrides": []
      },
      "gridPos": {
        "h": 8,
        "w": 12,
        "x": 0,
        "y": 40
      },
      "id": 410,
      "options": {
        "orientation": "auto",
        "reduceOptions": {
          "calcs": [
            "max"
          ],
          "fields": "",
          "values": false
        },
        "showThresholdLabels": false,
        "showThresholdMarkers": true
      },
      "pluginVersion": "8.5.1",
      "targets": [
        {
          "datasource": {
            "type": "influxdb",
            "uid": "yfZzmc1nk"
          },
          "query": "from(bucket: \"${Testrun}\")\r\n  |> range(start: v.timeRangeStart, stop: v.timeRangeStop)\r\n  |> filter(fn: (r) => r[\"_measurement\"] == \"ftp\")\r\n  |> filter(fn: (r) => r[\"_field\"] == \"duration\")\r\n  |> keep(columns: [\"source\", \"duration\", \"_time\", \"_value\"])\r\n  |> group(columns: [\"source\"])",
          "refId": "A"
        }
      ],
      "title": "max (duration)",
      "transformations": [],
      "type": "gauge"
    },
    {
      "datasource": {
        "type": "influxdb",
        "uid": "yfZzmc1nk"
      },
      "fieldConfig": {
        "defaults": {
          "color": {
            "mode": "thresholds"
          },
          "decimals": 3,
          "mappings": [],
          "min": 0,
          "thresholds": {
            "mode": "absolute",
            "steps": [
              {
                "color": "green"
              },
              {
                "color": "#EAB839",
                "value": 10
              },
              {
                "color": "red",
                "value": 25
              }
            ]
          },
          "unit": "ms"
        },
        "overrides": []
      },
      "gridPos": {
        "h": 8,
        "w": 12,
        "x": 12,
        "y": 40
      },
      "id": 451,
      "options": {
        "orientation": "auto",
        "reduceOptions": {
          "calcs": [
            "min"
          ],
          "fields": "",
          "values": false
        },
        "showThresholdLabels": false,
        "showThresholdMarkers": true,
        "text": {}
      },
      "pluginVersion": "8.5.1",
      "targets": [
        {
          "datasource": {
            "type": "influxdb",
            "uid": "yfZzmc1nk"
          },
          "query": "from(bucket: \"${Testrun}\")\r\n  |> range(start: v.timeRangeStart, stop: v.timeRangeStop)\r\n  |> filter(fn: (r) => r[\"_measurement\"] == \"ftp\")\r\n  |> filter(fn: (r) => r[\"_field\"] == \"duration\")\r\n  |> keep(columns: [\"source\", \"duration\", \"_time\", \"_value\"])\r\n  |> group(columns: [\"source\"])",
          "refId": "A"
        }
      ],
      "title": "min (duration)",
      "transformations": [],
      "type": "gauge"
    }
  ],
  "refresh": "5s",
  "schemaVersion": 36,
  "style": "dark",
  "tags": [
    "Overview"
  ],
  "templating": {
    "list": [
      {
        "allValue": "ai*",
        "current": {
          "selected": false,
          "text": "All",
          "value": "$__all"
        },
        "datasource": {
          "type": "influxdb",
          "uid": "yfZzmc1nk"
        },
        "definition": "buckets()",
        "hide": 0,
        "includeAll": true,
        "multi": true,
        "name": "Testrun",
        "options": [],
        "query": "buckets()",
        "refresh": 1,
        "regex": "/.*ai_dispatcher_mzd*./",
        "skipUrlSync": false,
        "sort": 2,
        "type": "query"
      },
      {
        "current": {
          "selected": true,
          "text": [
            "All"
          ],
          "value": [
            "$__all"
          ]
        },
        "datasource": {
          "type": "influxdb",
          "uid": "yfZzmc1nk"
        },
        "definition": "import \"influxdata/influxdb/v1\"\r\nv1.measurements(bucket: \"${Testrun}\")",
        "hide": 0,
        "includeAll": true,
        "multi": true,
        "name": "Measurement",
        "options": [],
        "query": "import \"influxdata/influxdb/v1\"\r\nv1.measurements(bucket: \"${Testrun}\")",
        "refresh": 2,
        "regex": "",
        "skipUrlSync": false,
        "sort": 1,
        "type": "query"
      },
      {
        "current": {
          "selected": true,
          "text": [
            "All"
          ],
          "value": [
            "$__all"
          ]
        },
        "datasource": {
          "type": "influxdb",
          "uid": "yfZzmc1nk"
        },
        "definition": "  from(bucket: \"${Testrun}\")\r\n  |> range(start: 0)\r\n  |> filter(fn: (r) => (r[\"_measurement\"] == \"${Measurement}\"))\r\n  |> keep(columns: [\"test_StartTime\"])\r\n  |> group()\r\n  |> distinct(column: \"test_StartTime\")\r\n  |> sort()",
        "hide": 0,
        "includeAll": true,
        "multi": true,
        "name": "Test_StartTime",
        "options": [],
        "query": "  from(bucket: \"${Testrun}\")\r\n  |> range(start: 0)\r\n  |> filter(fn: (r) => (r[\"_measurement\"] == \"${Measurement}\"))\r\n  |> keep(columns: [\"test_StartTime\"])\r\n  |> group()\r\n  |> distinct(column: \"test_StartTime\")\r\n  |> sort()",
        "refresh": 2,
        "regex": "",
        "skipUrlSync": false,
        "sort": 0,
        "type": "query"
      },
      {
        "current": {
          "selected": true,
          "text": [
            "All"
          ],
          "value": [
            "$__all"
          ]
        },
        "datasource": {
          "type": "influxdb",
          "uid": "yfZzmc1nk"
        },
        "definition": "  from(bucket: \"${Testrun}\")\n  |> range(start: 0)\n  |> filter(fn: (r) => (r[\"_measurement\"] == \"${Measurement}\" and r[\"test_StartTime\"] == \"${Test_StartTime}\"))\n  |> keep(columns: [\"source\"])\n  |> group()\n  |> distinct(column: \"source\")\n  |> sort()",
        "hide": 0,
        "includeAll": true,
        "multi": true,
        "name": "source",
        "options": [],
        "query": "  from(bucket: \"${Testrun}\")\n  |> range(start: 0)\n  |> filter(fn: (r) => (r[\"_measurement\"] == \"${Measurement}\" and r[\"test_StartTime\"] == \"${Test_StartTime}\"))\n  |> keep(columns: [\"source\"])\n  |> group()\n  |> distinct(column: \"source\")\n  |> sort()",
        "refresh": 2,
        "regex": "",
        "skipUrlSync": false,
        "sort": 0,
        "type": "query"
      },
      {
        "current": {
          "selected": true,
          "text": [
            "All"
          ],
          "value": [
            "$__all"
          ]
        },
        "datasource": {
          "type": "influxdb",
          "uid": "yfZzmc1nk"
        },
        "definition": "  from(bucket: \"${Testrun}\")\n  |> range(start: 0)\n  |> filter(fn: (r) => (r[\"_measurement\"] == \"${Measurement}\" and r[\"test_StartTime\"] == \"${Test_StartTime}\"))\n  |> keep(columns: [\"destination\"])\n  |> group()\n  |> distinct(column: \"destination\")",
        "hide": 0,
        "includeAll": true,
        "multi": true,
        "name": "destination",
        "options": [],
        "query": "  from(bucket: \"${Testrun}\")\n  |> range(start: 0)\n  |> filter(fn: (r) => (r[\"_measurement\"] == \"${Measurement}\" and r[\"test_StartTime\"] == \"${Test_StartTime}\"))\n  |> keep(columns: [\"destination\"])\n  |> group()\n  |> distinct(column: \"destination\")",
        "refresh": 1,
        "regex": "",
        "skipUrlSync": false,
        "sort": 0,
        "type": "query"
      }
    ]
  },
  "time": {
    "from": "1970-01-01T00:59:07.000Z",
    "to": "1970-01-01T01:45:34.000Z"
  },
  "timepicker": {},
  "timezone": "utc",
  "title": "MzD - Overview",
  "uid": "7okBL_X7k",
  "version": 2,
  "weekStart": ""
}
`
</details>

or in File:

[mzd_dashboard_config.txt](https://github.com/grafana/grafana/files/8861104/mzd_dashboard_config.txt)

**Comment by svogel123** [comment]
I have setup a new server with grafana 9.0, but the isssue is still there.

On setup the data source shows me the following:
![2022-06-09_07h45_19](https://user-images.githubusercontent.com/104064222/172776862-55b39925-0633-4870-884f-b85220e36bed.png)

**Comment by yesoreyeram** [comment]
In grafana 9 settings page, doesn't the number match the total number of buckets?  It should match the numbers with the same as the curl request results provided in the previous comment.

**Comment by svogel123** [comment]
> In grafana 9 settings page, doesn't the number match the total number of buckets? It should match the numbers with the same as the curl request results provided in the previous comment.

not on my server ;)

can we setup an session where you can look on my hardware to reproduce the issue?

**Comment by thomasbromkapsdata** [comment]
Hi,
I have the same issue.
I have more than 18 buckets but Grafana only see 18 of them.
When I set up the datasource it say "18 bucket found" and if I use "bucket()" in flux for a variable or whatever it is only 18.
However, I am able to query the shadow buckets (so it is an issue with bucket() but not a problem with the datasource itself)
More fun:
-if I delete a bucket in Influx, another one take the place of the deleted one to tay to 18.
-if I set up a Token in influx to read only a bucket who is not from the 18 chosen and put it in the datasource, the datasource failed to "save and test"
-if a set up a Token to access 3 of the 18 chosen and 2 of the other buckets, it show "3 buckets found".


Environment:

    Grafana version: v9.0.4 
    OS Grafana is installed on: Debian 10
    User OS & Browser: Windows 10 with Firefox
    InfluxDB v2.3.0
